contractcourt/chain_arbitrator.go

```diff
@@ -770,20 +778,69 @@ func (c *ChainArbitrator) Start() error {
 
 	// Launch all the goroutines for each arbitrator so they can carry out
 	// their duties.
+	// Set a timeout for the group so we don't wait indefinitely if one
+	// startup function blocks.
+	ctx, cancel := context.WithTimeout(
```

Comment:
I'm not sure the cancel and error group here gives us much.
This would of course be different if we passed an actual context into `arbitrator.Start()` that would then also cancel the blocking behavior. But that's currently not the case and probably a much bigger diff.

So then the error group also doesn't help us, because canceling the context won't stop new goroutines to be spawned.

So here's the approach I came up with (potentially not perfect yet either, mostly to show you what I mean):

```
	// Launch all the goroutines for each arbitrator so they can carry out
	// their duties.
	// Set a timeout for the group so we don't wait indefinitely if one
	// startup function blocks.
	c.Lock()
	numArbitrators := len(c.activeChannels)
	startupTimeout := time.After(c.cfg.StartupTimeout)
	startErrors := make(chan error, numArbitrators)

	for _, arbitrator := range c.activeChannels {
		startState, ok := startStates[arbitrator.cfg.ChanPoint]
		if !ok {
			c.Unlock()

			stopAndLog()

			return fmt.Errorf("arbitrator: %v has no start state",
				arbitrator.cfg.ChanPoint)
		}

		// Create a non-blocking goroutine for the actual Start call.
		go func(a *ChannelArbitrator) {
			startErrors <- a.Start(startState)
		}(arbitrator)
	}

	// Wait for the group to finish and if an error occurred we trigger a
	// graceful shutdown of LND.
	go func() {
		defer c.Unlock()
		
		var numStarted int
		for {
			select {
			case startErr := <-startErrors:
				if startErr != nil {
					stopAndLog()

					log.Criticalf("ChainArbitrator failed "+
						"to start all channel "+
						"arbitrators: %v", err)
					
					return
				}
				
				numStarted++
				
				// We've successfully started all arbitrators.
				if numStarted == numArbitrators {
					return
				}

			case <-startupTimeout:
				stopAndLog()

				log.Criticalf("timeout waiting for channel " +
					"arbitrator start")

				return

			case <-c.quit:
				err = ErrChainArbExiting
			}
		}

	}()
```

Main changes:
 - Instead of spawning a goroutine within the error group goroutine, we just spawn one per arbitrator and collect into a single channel.
 - We don't use a cancel context because it can't be passed into `arbitrator.Start()` yet anyway. So a simple timeout is enough.
 - Because we now operate on the arbitrators outside of the scope of `Start()` (within goroutines), we probably also need to block changes to those arbitrators until we're done. That's why I added the `c.Lock()` and corresponding unlocks. But not sure if those could lead to new deadlocks, so we need to look at those very carefully.